runtime/caml/fiber.h

```diff
@@ -99,7 +103,47 @@ struct c_stack_link {
   struct c_stack_link* prev;
 };
 
-#define NUM_STACK_SIZE_CLASSES 5
+/* Overview of the native stack switching primitives for effects
+ *
+ * For an understanding of effect handlers in OCaml please see:
+ *  Retrofitting Effect Handlers onto OCaml, KC Sivaramakrishnan, et al.
+ *  PLDI 2021
+ *
+ * In native compilation the stack switching primitives Prunstack,
+ * Pperform, Preperform and Presume make use of corresponding functions
+ * implemented in the assembly files for an architecture (such as
+ * runtime/amd64.S).
```

Comment:
For completeness, you could mention what happens in bytecode?  I.e. that these primitives are mapped to instructions of the abstract machine.

Reply:
Yes, good idea.